heracles/dices/jackknife.py

```diff
@@ -43,6 +43,11 @@ def jackknife_cls(data_maps, vis_maps, jk_maps, fields, nd=1):
     """
     if nd < 0 or nd > 2:
         raise ValueError("number of deletions must be 0, 1, or 2")
+    # Set _mask to array of -1 if input jkmap is None
+    # This is for eg. XC with systematics maps 
+    for key_mask in jk_maps.keys():
```

Comment:
You are changing the input dictionary here.

```suggestion
    # This is for eg. XC with systematics maps
    # Work on a mutable copy of the input
    jk_maps = dict(jk_maps)
    for key_mask in jk_maps.keys():
```